azurerm/internal/services/securitycenter/resource_arm_security_center_subscription_pricing.go

```diff
@@ -86,15 +124,17 @@ func resourceArmSecurityCenterSubscriptionPricingRead(d *schema.ResourceData, me
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	resp, err := client.GetSubscriptionPricing(ctx, securityCenterSubscriptionPricingName)
+	resource_type := d.Get("resource_type").(string)
+
+	resp, err := client.Get(ctx, resource_type)
 	if err != nil {
 		if utils.ResponseWasNotFound(resp.Response) {
-			log.Printf("[DEBUG] Security Center Subscription was not found: %v", err)
+			log.Printf("[DEBUG] %v Security Center Subscription was not found: %v", resource_type, err)
 			d.SetId("")
 			return nil
 		}
 
-		return fmt.Errorf("Error reading Security Center Subscription pricing: %+v", err)
+		return fmt.Errorf("Error reading %v Security Center Subscription pricing: %+v", resource_type, err)
```

Comment:
minor change here for display
```suggestion
		return fmt.Errorf("Error reading %q Security Center Subscription pricing: %+v", resource_type, err)
```